.golangci.yml

```diff
@@ -126,5 +126,10 @@ issues:
     - path: _test\.go
       text: "parameter.*seems to be a control flag, avoid control coupling"
 
+    - path: (^agent/|^cmd/|^config/|^filter/|^internal/|^logger/|^metric/|^models/|^selfstat/|^testutil/|^plugins/serializers/)
+      text: "imports-blacklist: should not use the following blacklisted import: \"log\""
+      linters:
+        - revive
+
```

Comment:
I did a quick `make lint` and there were only 21 instances of this message I'd rather you continue commenting on the 21 or less that can not actually be fixed.

Making large exceptions is something we should avoid unless we have a really good reason.

Reply:
@powersj 
Reason to have this large exception is mentioned in the description of this PR :)

`make lint | grep "blacklisted"` (in this PR) gave me 7 findings, all of them need to be addressed one by one (I didn't address them in my previous PRs, because they are not so easy to fix and maybe some collective decision needs to be made)

Reply:
Ah I guess I had assumed you would put the `//no lint:revive` anywhere this is actually used, negating the reason to put a large exception.

Reply:
For master branch there is indeed 21 findings right now.

But there is a document over here: https://github.com/influxdata/telegraf/blob/master/docs/developers/LOGGING.md#agent-logging
in which there is a statement:
```
In other sections of the code it is required to add the log level and module manually:
log.Printf("E! [agent] Error writing to %s: %v", output.LogName(), err)
```

Reply:
My fear is that we put this exception in place, and then we ignore it all time since it won't be noisy. Ideally, this should be fixed by creating a generic logger that can be used anywhere in the code in the event we ever wanted to change the log format or other settings. The reality of us doing that is slim, so I guess the exceptions are ok, but not ideal.

As far as `plugins/serializers`, why is that included?

Reply:
When generic logger is ready, we will remove this exception :)

To inject `telegraf.Logger` into plugin `SetLoggerOnPlugin` function from `models/log.go` is used. I think that it is not called for serializers. Or am I missing something?

Reply:
> When generic logger is ready, we will remove this exception :)

Please file a bug to ensure this is not forgotten with reference to this PR, thanks!

> To inject telegraf.Logger into plugin SetLoggerOnPlugin function from models/log.go is used. I think that it is not called for serializers. Or am I missing something?

Can you give me an example? This should be like any other plugin per the docs you linked to. Where possible, add `Log telegraf.Logger `toml:"-"`` to the struct, and call the logger from there. Like other plugins, there may be exceptions where functions are outside a struct type.

Reply:
> Please file a bug to ensure this is not forgotten with reference to this PR, thanks!

Done: https://github.com/influxdata/telegraf/issues/10212

> Can you give me an example? This should be like any other plugin per the docs you linked to. Where possible, add `Log telegraf.Logger `toml:"-"`` to the struct, and call the logger from there. Like other plugins, there may be exceptions where functions are outside a struct type.

Exactly. And functionality you mentioned is done by `SetLoggerOnPlugin` function from `models/log.go`
![image](https://user-images.githubusercontent.com/17336997/144582373-1ee5f044-59c2-49ff-973b-74fb967d0964.png)
And this function is being called for every `aggregator`, `input`, `output`, `parser` and `processor` but not for `serializer`.

Reply:
ok - I'd rather fix this via 10212 then.